### PR TITLE
add generic crt

### PR DESCRIPTION
**Pull request by tthsqe12**
I don't care about the arrangement of the args in crt(r1, m1, r2, m2)

### CODE REVIEW

File: src/algorithms/GenericFunctions.jl [hunk 1]
```diff
@@ -301,18 +301,85 @@ function gcdinv(a::T, b::T) where T <: RingElem
...
+end
+
+function _crt_stub(r1::T, m1::T, r2::T, m2::T; check::Bool=true) where T <: RingElement
+    return _crt_with_lcm_stub(r1, m1, r2, m2; check=check)[1]
+end
+
+@doc Markdown.doc"""
+    crt(r1::T, m1::T, r1::T, m2::T; check::Bool=true) where T <: RingElement
 
-# TODO: Move from CRT from Hecke to AbstractAlgebra?
```
Comment (thofma) [inline src/algorithms/GenericFunctions.jl:305]: Why remove this comment? It is still missing the version using the tree.

Comment (tthsqe12) [reply src/algorithms/GenericFunctions.jl:305]: ??
There is no crt_env here

Comment (tthsqe12) [reply src/algorithms/GenericFunctions.jl:305]: How julia does the "tree" is implementation defined

Comment (thofma) [reply src/algorithms/GenericFunctions.jl:305]: The CRT stuff from Hecke is not moved to AA, so why remove the comment?

Comment (fieker) [reply src/algorithms/GenericFunctions.jl:305]: Will this always be correct for Int? At least if product of moduli < typemax?

Comment (tthsqe12) [reply src/algorithms/GenericFunctions.jl:305]: ok

**Comment by tthsqe12** [comment]
The idea here is that there are stubs that actually do the useful stuff. Then, Nemo.ZZ can call the crt_with_lcm stub to get (a,m) and return divrem(a,m)[1] or nfdivrem(a,m)[2] depending on the sign.

**Comment by thofma** [comment]
This is breaking something in Nemo: https://github.com/Nemocas/AbstractAlgebra.jl/runs/7539960434?check_suite_focus=true

Do you want this here breaking?

**Comment by tthsqe12** [comment]
I actually had this in the back of my mind: Nemo's gcdinv(a,b) current has unnecessary restrictions on the inputs.

**Comment by tthsqe12** [comment]
Is there a way I can detect Nemo.ZZ in the AA code (<:Integer doesn't work) to get around Nemo's problems?

**Comment by thofma** [comment]
No

**Comment by tthsqe12** [comment]
It should be possible for this to be non-bresking. Will investigate failures tmr.

**Comment by tthsqe12** [comment]
The only failures here are:
- a strange one in GB on macos. It looks like some evil random examples popped up and blew up the memory.
- a silly one related to doc test
So, I would say it is non-breaking now.

**Pull request merged**